src/components/blocks/software/Code/ApiKeyIndicator.tsx

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
+import React, { useState } from 'react';
+import Icon from '@ably/ui/core/Icon';
 
-const tooltip =
-  'This code example uses a temporary key that is rate limited and expires in 4 hours. Sign in to Ably to use your API keys instead.';
-
-const APIKeyIndicator = () => (
-  <div className="docs-api-key-label" title={tooltip}>
-    Api Key: Demo Only
-  </div>
-);
+const APIKeyIndicator = ({ tooltip }: { tooltip: string }) => {
+  const [tooltipHover, setTooltipHover] = useState<boolean>(false);
+  return (
+    <div
+      className="flex flex-row w-full justify-start mt-2"
+      onMouseOver={() => setTooltipHover(true)}
+      onMouseOut={() => setTooltipHover(false)}
```

Comment:
The inline handlers would cause an extra render that we can avoid by passing function references.